registry/storage/cache/redis/redis.go

```diff
@@ -268,7 +270,7 @@ func (rsrbds *repositoryScopedRedisBlobDescriptorService) repositoryBlobSetKey(c
 }
 
 func cacheKeyPrefix(ctx context.Context) string {
-	prefix, ok := ctx.Value("distributionCacheKeyPrefix").(string)
+	prefix, ok := ctx.Value(ContextCacheKeyPrefix).(string)
 	if ok && prefix != "" {
```

Comment:
This would be the idiomatic way of doing this.  The constant is fine AFAIC though. 
```go
type contextKey string
const contextKeyPrefix = contextKey("distributionCacheKeyPrefix")

func ContextKeyPrefix(ctx context.Context) string {
	prefix, ok := ctx.Value(contextKeyPrefix).(string)
	if ok && prefix != "" {
		return prefix + "::"
	}
       return ""
}

func WithContextKeyPrefix(ctx context.Context, prefix string) context.Context {
	return context.WithValue(ctx, contextKeyPrefix, prefix)
}
```